### PR TITLE
display: add support for LPM0XX display series by JDI

### PR DESCRIPTION
**Pull request by protobits**
These displays are very similar to Sharp's memory LCD's but with slight differences in protocol and color format. I added a separate driver to avoid spaghetti code but in case you feel it is better to create one driver from both I can rework it.

This addition required extending the list of supported color formats to include a four-bit (r,g,b,dummy) format, which is used by LPM0126C, the one I'm using. The display also supports three-bit (r,g,b) and one-bit monochrome but for now I'm focused on color support and the three bit format really complicates memory layout. 

I also extended the display sample to support this new format and also improved it to account for SCREEN_INFO_X_ALIGNMENT_WIDTH display capability, which is actually used in this series of displays (as well as Sharp's).
This required quite some rework to the sample but I couldn't find a better way which wouldn't be an even larger rewrite or potential change in how the sample worked.

I've opened this as a draft since I would like to get your input on the change. Also, this is my first contribution to Zephyr so I expected to mess something up in the process.

PS: should I add myself to copyright lines in affected files?

### CODE REVIEW

**Comment by carlescufi** [comment]
> PS: should I add myself to copyright lines in affected files?

It really depends on the scope of the changes made to those files. Feel free to add yourself if you think the changes are substantial enough to warrant it.

File: drivers/display/display_lpm0xx.c [hunk 1]
```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (c) 2020 Rohit Gujarathi
```
Comment (carlescufi) [inline drivers/display/display_lpm0xx.c:2]: Where does this code originate from?

**Comment by carlescufi** [comment]
@protobits thanks for the PR.

You seem to have some checkpatch failures: https://github.com/zephyrproject-rtos/zephyr/pull/36212/checks?check_run_id=2820592719

File: drivers/display/display_lpm0xx.c (hunk 1 above)
Comment (protobits) [reply drivers/display/display_lpm0xx.c:2]: It was originally copied from ls0xx.c file on the same directory.

**Comment by protobits** [comment]
> @protobits thanks for the PR.
> 
> You seem to have some checkpatch failures: https://github.com/zephyrproject-rtos/zephyr/pull/36212/checks?check_run_id=2820592719

Thanks, I fixed style errors locally using checkpatch.

**Pull request closed without merging**